crates/bevy_math/src/angle/mod.rs

```diff
@@ -0,0 +1,312 @@
+    /// Returns the sine and cosine of the angle.
+    fn sin_cos(self) -> (f32, f32);
+
+    /// Returns the absolute angle.
+    #[inline]
+    fn abs(self) -> Self {
+        Self::new(self.value().abs())
+    }
+
+    /// Returns the minimum of the two angles, ignoring NaN.
+    #[inline]
+    fn min(self, other: Self) -> Self {
+        Self::new(self.value().min(other.value()))
+    }
+
+    /// Returns the maximum of the two angles, ignoring NaN.
+    #[inline]
+    fn max(self, other: Self) -> Self {
+        Self::new(self.value().max(other.value()))
+    }
+
+    /// Normalizes the angle to be within the range of `[0, 360)` degrees or `[0, 2π)` radians.
+    #[inline]
+    fn normalized(self) -> Self {
+        if self < Self::FULL && self > Self::ZERO {
+            self
+        } else {
+            let remainder = self % Self::FULL;
+            if remainder >= Self::ZERO {
+                remainder
+            } else {
+                remainder + Self::FULL
+            }
+        }
+    }
+
+    /// Returns the largest integer angle less than or equal to `self`.
+    #[inline]
+    fn floor(self) -> Self {
+        Self::new(self.value().floor())
+    }
+
+    /// Returns the smallest integer angle greater than or equal to `self`.
+    #[inline]
+    fn ceil(self) -> Self {
+        Self::new(self.value().ceil())
+    }
+
+    /// Returns the nearest integer to `self`. If a value is half-way
+    /// between two integers, round away from `0.0`.
+    #[inline]
+    fn round(self) -> Self {
+        Self::new(self.value().round())
+    }
+
+    /// Adds the given `angle` to `self`, returning the inferred type of angle.
+    fn add_angle<T: Angle, Output: Angle>(self, angle: T) -> Output
+    where
+        Output: From<Self> + From<T>,
+    {
+        Output::from(self) + Output::from(angle)
+    }
+
+    /// Subtracts the given `angle` from `self`, returning the inferred type of angle.
+    fn sub_angle<T: Angle, Output: Angle>(self, angle: T) -> Output
+    where
+        Output: From<Self> + From<T>,
+    {
+        Output::from(self) - Output::from(angle)
+    }
+
+    /// Multiplies `self` by the given `angle`, returning the inferred type of angle.
+    fn mul_angle<T: Angle, Output: Angle>(self, angle: T) -> Output
+    where
+        Output: From<Self> + From<T>,
+    {
+        Output::from(self) * Output::from(angle)
+    }
+
+    /// Divides `self` by the given `angle`, returning the inferred type of angle.
+    fn div_angle<T: Angle, Output: Angle>(self, angle: T) -> Output
+    where
+        Output: From<Self> + From<T>,
+    {
+        Output::from(self) + Output::from(angle)
+    }
+
+    /// Computes the remainder of `self` divided by the given `angle`, returning the inferred type of angle.
+    fn rem_angle<T: Angle, Output: Angle>(self, angle: T) -> Output
+    where
+        Output: From<Self> + From<T>,
+    {
+        Output::from(self) % Output::from(angle)
+    }
+}
+
+impl PartialEq<Radians> for Degrees {
```

Comment:
This is `d == r.to_degrees()`. Below is `r == d.to_radians()`. My gut feeling is that I can find two floating point values such that `d == r` is true, and `r == d` is false, because of the different conversion errors. In other words, this breaks the symmetry of `PartialEq`.

Reply:
Good catch. I'll change it so that it just does `r.to_degrees()` for both since it might be slightly cheaper than `to_radians`, one multiplication vs. a multiplication and a division, although it'd probably get optimized by the compiler anyway. Not sure if one is more precise than the other, but it'd probably be a negligible difference either way.

Reply:
By "a multiplication and a division" I meant that `to_radians` is this

```rust
pub fn to_radians(self) -> f32 {
    let value: f32 = consts::PI;
    self * (value / 180.0f32)
}
```

but that surely gets optimized by the compiler to be just a multiplication like I suggested. So yeah, the performance should be equivalent.